### PR TITLE
Expose app_manifest for configuration

### PR DESCRIPTION
**Pull request by ryanwholey**
Adds the ability to configure the app manifest for the proposed deployment. The caller should be able to pass multiple values here by using a [job matrix](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix).

Changes:
- Moves app_manifest out to an action input
- Move context and config up to index.js so that it's fairly easy to test deploy.js
- Add tests for deploy.js

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -24,6 +24,10 @@ inputs:
     description: Default CI URL prefix, used to create the CI build URL in the Harbormaster package 
     required: false
     default: https://app.circleci.com/pipelines/github/
+  deployment_manifests:
```
Comment (bendrucker) [inline action.yml:27]: `app_manifests`?

**Review by bendrucker — COMMENTED** [review]
Is this even necessary, or could the caller just use a job matrix?

https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix

This occurred to me when working on CircleCI configs, and I'm definitely going to be able to make use of matrixes there to remove a bit of the existing duplication.

> Side note, i'm thinking about removing the update dist on pull_request, its going to show up in pull reuqests and its annoying to have to pull down commits when iterating on a pr

Indeed, just means removing branch protection. Which is fine.

File: src/deploy.test.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  deploymentManifests: 'test/fixtures/app1.yaml',
+  version: '124cb6d97d86ed4b93eccc95a8ce4ff58a24f843-test',
+  ciUrlPrefix: 'https://app.circleci.com/pipelines/github/',
+  gitRef: '124cb6d97d86ed4b93eccc95a8ce4ff58a24f843',
+  branch: 'feature',
+  trigger: '/deploy',
+  workspace: '.'
+}, extensions)
+
+describe('deploy', () => {
```
Comment (bendrucker) [inline src/deploy.test.js:48]: Also, would be nice to use nock here and generally avoid Sinon where possible

Comment (ryanwholey) [reply src/deploy.test.js:48]: My thought is that nock belongs more around testing the harbormaster library itself and in deploy.test.js testing that the client was called correctly. Sinon isn't really doing anything here aside from decorating the test client to assert calls.

**Comment by ryanwholey** [comment]
ah cool! I hadn't seen matrices in GHAs before, ill take a look

File: src/deploy.test.js (hunk 2 above)
Comment (bendrucker) [reply src/deploy.test.js:48]: Yeah on closer inspection, using Sinon to construct your own mock is fine. It's really just the stubbing of shared objects that's bad. Even using stubs to construct the mock methods is fine, as long as the object is a mock you constructed and not part of the source code.

**Comment by bendrucker** [comment]
Yeah! Matrixes get extra fun when it's more than `1:`. But it's handy even for the basic case of multiple values for a single variable. There's an important/valuable behavioral change too: distinct jobs are spawned in parallel. If one fails, the others still proceed to their terminal state. This prevents users from relying on deterministic ordering and allows some work to proceed even if part of the matrix fails.

**Comment by bendrucker** [comment]
The classic case for matrixes would be something like this:

```yaml
jobs:
  test:
    name: ${{ matrix.os }} (node ${{ matrix.node }})
    runs-on: ${{ matrix.os }}
    strategy:
      matrix:
        os: [ubuntu-latest, windows-latest]
        node: ['12', '14']
    steps:
    - uses: actions/checkout@v2
    - uses: actions/setup-node@v2
      with:
        node-version: ${{ matrix.node }}
```

Test on Windows and Linux, Node.js 12 and 14, for a 2x2 matrix

**Review by bendrucker — APPROVED** [review]
Dope!

**Pull request merged**